it/common/src/main/java/org/apache/beam/it/common/PipelineLauncher.java

```diff
@@ -0,0 +1,422 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.it.common;
+
+import static org.apache.beam.it.common.utils.PipelineUtils.createJobName;
+
+import com.google.api.services.dataflow.model.Job;
+import com.google.auto.value.AutoValue;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableMap;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableSet;
+
+/** Client for working with Cloud Dataflow. */
+public interface PipelineLauncher {
+  /** Enum representing Apache Beam SDKs. */
+  enum Sdk {
+    JAVA("JAVA"),
+    PYTHON("PYTHON"),
+    GO("GO");
+
+    private final String text;
+
+    Sdk(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public String toString() {
+      return text;
+    }
+  }
+
+  /** Enum representing known Dataflow job states. */
+  enum JobState {
```

Comment:
Aren't these already in the SDK?

Reply:
Not all of them. These represent the `JobState` values for dataflow. States like `DRAINING` are not in the SDK?